src/ItemLinkManagement/Public/New-ItemLink.ps1

```diff
@@ -33,14 +33,21 @@ function New-ItemLink {
                 if (!$item.LinkType) {
                     throw "Existing item '$_path' is not a HardLink, Junction, or SymbolicLink."
                 }
-                if (($item.LinkType -eq $ItemType) -and ($item.Target -eq $_value)) {
-                    "Matching item '$_path' already exists. Skipping" | Write-Verbose
-                    return
+                if (!$Force) {
+                    if (($item.LinkType -eq $ItemType) -and ($item.Target -eq $_value)) {
+                        "Matching item '$_path' already exists. Skipping" | Write-Verbose
+                        return
+                    }
                 }
+                # New-Item -Force does not override an existing junction, hence the need to remove the existing junction
                 if ($ItemType -eq 'Junction') {
-                    if ($item.LinkType -ne $ItemType) {                             # New-Item -Force does not work for junctions, hence the need to remove the existing item
-                        "Itemtype specified as 'Junction'. Removing item '$($item.FullName)' of different item type '$($item.LinkType)'" | Write-Verbose
-                        $item.Delete()                                              # Remove-Item -Force and -Confirm:$false do not suppress confirmation for removal if items exists within symlink or junction target
+                    # Powershell 5 requires a special way to remove a SymbolicLink, see: https://stackoverflow.com/a/63172492
+                    # Remove-Item -Force and -Confirm:$false do not suppress confirmation for removal if items exists within symlink or junction target
+                    if ($PSVersionTable.PSVersion.Major -le 5 -and $item.Attributes -match 'ReparsePoint') {
+                        $item.Delete()
+                    }else {
+                        # Remove the existing junction
+                        $item | Remove-Item -Force
```

Comment:
Alright so from my understanding both `junction` and `symboliclink` items contain `ReparsePoint` under `.attributes`. Do these changes imply `.Delete()` the "special" way for removing symbolic links on PS `5.*`?

However based on my inline comment `Remove-Item -Force and -Confirm:$false do not suppress confirmation for removal if items exists within symlink or junction target` I think `.Delete()` is also required regardless of PS `5.*` **when** symbolic or junction targets contain items. I believe this is where the present scenarios in the unit tests fail to cover - where they are items within target directories of those symlinks. As the comment describes, `Remove-Item -Force` nor `-Confirm:$false` do anything to suppress removal of such links. As such `$item.Delete()` should only be used and `Remove-Item -Force` only for perhaps hardlinks. @leojonathanoh

Reply:
> in cases where junction or symboliclink targets contain items

junction or symboliclink do not contain items, but their source of truth does. I'm not sure how u arrived at the conclusion in that inline comment, but I think it might be a fluke discovery.

Reply:
@joeltimothyoh according to your prediction the tests fail to cover a scenario. Perhaps we can try adding one item inside a junction or symlink, and according to your expectation the test should fail.

Reply:
# `junction`

```
$ New-ItemLink "D:\test\junction" -itemtype junction -Value "C:\test\target" -Verbose -Force
```

## `powershell 5.1`

### `Remove-Item`

```
New-ItemLink : D:\test\junction is an NTFS junction point. Use the Force parameter to delete or modify this object.
```

### `Remove-Item -Force`

```
Confirm
The item at Microsoft.PowerShell.Core\FileSystem::D:\test\junction has children and the Recurse parameter was not specified. If you continue, all children will be removed with the item. Are you sure you want to continue?
[Y] Yes [A] Yes to All [N] No [L] No to All [S] Suspend [?] Help (default is "Yes"):
```

### `Remove-Item -Force -Recurse`

```
    Directory: D:\test


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----l          2/8/2021     23:35                junction
```

## `pwsh 6.2.4`

### `Remove-Item`

```
    Directory: D:\test


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----l          2/8/2021     23:35                junction
```

### `Remove-Item -Force`

```
    Directory: D:\test


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----l          2/8/2021     23:35                junction
```
### `Remove-Item -Force -Recurse`

```
    Directory: D:\test


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----l          2/8/2021     23:35                junction
```

@leojonathanoh

Reply:
So it looks like all that is necessary is specifying `Remove-Item` with both `-Force` and `Recurse` for `junction` items. Appears to do the same as `.Delete()`. @leojonathanoh

Reply:
@joeltimothyoh basically as we already expect, we see that only symboliclinks need to use `.Delete()`

Reply:
But based on the present code `$item.Delete()` and `$item | Remove-Item -Force` only apply to `junction` items as they fall are under the following if-else block https://github.com/theohbrothers/ItemLinkManagement/pull/11#discussion_r681086562 @leojonathanoh

Reply:
Also it is correct that only `junction` items require deletion before creation. The other links can be recreated with `New-Item -Force`. PS `5.*` but requires escalation for creation and deletion of `symboliclink items.` @leojonathanoh

Reply:
And if indeed `$item.Delete()` is all that is required for removal of `junction` items regardless of PS version, `Remove-Item -Force -Recurse` is unneeded. In fact, no changes to `src/ItemLinkManagement/Public/New-ItemLink.ps1` is even necessary. @leojonathanoh

Reply:
@joeltimothyoh to be clear ` $item.Delete()` is required only for symboliclink deletion on PSv5. Everything else can be removed using `Remove-Item`.

Reply:
Generally `Remove-Item -Recurse -Force` is a very dangerous command and it should only be used only if there's no other way to achieve what one wants.

Reply:
> @joeltimothyoh to be clear $item.Delete() is required only for symboliclink deletion on PSv5. Everything else can be removed using Remove-Item.

@leojonathanoh  Not true. Admin privileges is what's necessary for managing symlinks on `powershell`.

# `symboliclink `

* Command: `New-ItemLink "D:\test\symboliclink" -ItemType symboliclink -Value "C:\test\target" -Verbose -Force`
* `C:\test\target` with files.

## `powershell 5.1`

### `Remove-Item`

```
New-ItemLink : D:\test\junction is an NTFS junction point. Use the Force parameter to delete or modify this object.
```

### `Remove-Item -Force`

```
    Directory: D:\test


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----l          3/8/2021     00:07                symboliclink
```

### `Remove-Item -Force -Recurse`

```
    Directory: D:\test


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----l          3/8/2021     00:07                symboliclink
```

Reply:
@joeltimothyoh based on https://github.com/theohbrothers/ItemLinkManagement/pull/11#discussion_r681099231, i will consider the PR good to go

Reply:
> @joeltimothyoh to be clear $item.Delete() is required only for symboliclink deletion on PSv5. Everything else can be removed using Remove-Item.

Because `symboliclink` items can be recreated with `New-Item -Force` regardless of PS version, removing them is unnecessary. @leojonathanoh

Reply:
@joeltimothyoh ultimately we are arriving at a desired state. If it takes removal and then creation, we still arrive at a desired state. 

So it is more explicity in the code, that right before the `New-Item` command, the existing Link item is deleted. I prefer this explicit behavior for understandability.